aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1659,65 +1672,89 @@ AT_ERROR("cholesky: MAGMA library not found in "
     // which concisely is equal to batch_size % batch_limit
     if (batch_size % batch_limit != 0) {
       magmaCholeskyBatched<scalar_t>(
-        uplo, n, &self_array[mini_idx], lda, &info_array[mini_idx], batch_size % batch_limit, magma_queue);
-    }
-
-    for (int64_t i = 0; i < batch_size; i++) {
-      infos[i] = info_array[i];
+        uplo, n, &self_array[mini_idx], lda, &info_data[mini_idx], batch_size % batch_limit, magma_queue);
     }
   }
 #endif
 }
 
-Tensor _cholesky_helper_cuda_magma(const Tensor& self, bool upper) {
-  std::vector<int64_t> infos(batchCount(self), 0);
-
-  Tensor result;
-  if (self.dim() > 2) {
+void cholesky_helper_magma(const Tensor& input, bool upper, const Tensor& info) {
+  Tensor result = input;
+  if (input.dim() > 2) {
     // MAGMA's batched cholesky operator has an off-by-one error causing IMA
     // (see https://github.com/pytorch/pytorch/issues/42666). This code is based
     // on the #cloneBatchedColumnMajor function however it pads the input with
     // one extra element utilizing the fact that the resize_as_ method preserves
     // the storage even if it's larger than the new sizes. This way if MAGMA
     // reads off bounds it will still be valid user memory.
-    const Tensor input = upper ? self : self.transpose(-1, -2);
     result = at::empty(input.numel() + 1, input.options());
-    result.resize_as_(input).copy_(input).transpose_(-1, -2);
-  } else {
-    result = cloneBatchedColumnMajor(upper ? self.transpose(-1, -2) : self);
+    result.resize_as_(input).transpose_(-2, -1);
+    TORCH_INTERNAL_ASSERT(result.transpose(-2, -1).is_contiguous());
+    result.copy_(input);
   }
 
   AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(
-      self.scalar_type(), "cholesky_cuda", [&] {
-        apply_cholesky<scalar_t>(result, false, infos);
-      });
+    input.scalar_type(), "cholesky_cuda", [&] {
+      apply_cholesky<scalar_t>(result, upper, info);
+    });
 
-  if (self.dim() > 2) {
-    batchCheckErrors(infos, "cholesky_cuda");
-  } else {
-    singleCheckErrors(infos[0], "cholesky_cuda");
+  if (input.dim() > 2) {
+    // if upper=true we need to tranpose and conjugate the result tensor
+    // because the cholesky decomposition is stored in the lower triangular part
+    if (upper) {
+      input.copy_(result.conj().transpose(-2, -1));
+    } else {
+      input.copy_(result);
+    }
   }
-
-  return upper ? result.transpose_(-1, -2) : result;
 }
 
 // Todo: cusolverDnXpotrfBatched has some numerical issue and is not used
 //     here. Batched cholesky is dispatched to magma.
 //     We will switch to cusolverDnXpotrfBatched after the issue is fixed.
 //     See https://github.com/pytorch/pytorch/issues/53879.
 Tensor _cholesky_helper_cuda(const Tensor& self, bool upper) {
+  auto info_shape = IntArrayRef(
+    self.sizes().cbegin(), self.sizes().cend() - 2); // self.shape[:-2]
+  Tensor self_working_copy = cloneBatchedColumnMajor(self);
```

Comment:
can you please leave a comment here that clone is called because legacy cholesky doesn't do clone before?